tests/lib/at_cmd_parser/at_params/src/main.c

```diff
@@ -0,0 +1,448 @@
+}
+
+static void test_params_put_get_int_teardown(void)
+{
+	at_params_list_free(&test_list);
+}
+
+static void test_params_put_get_string_setup(void)
+{
+	at_params_list_init(&test_list, TEST_PARAMS);
+	at_params_int_put(&test_list, 0, 1);
+}
+
+static void test_params_put_get_string(void)
+{
+	const char test_str[] = "Test, 1, 2, 3";
+	char test_buf[32];
+	size_t test_buf_len = sizeof(test_buf);
+
+	zassert_equal(-EINVAL, at_params_string_put(NULL, 1,
+					test_str, sizeof(test_str)),
+		      "String put should return -EINVAL");
+
+	zassert_equal(-EINVAL, at_params_string_put(&test_list, 1,
+					NULL, sizeof(test_str)),
+		      "String put should return -EINVAL");
+
+	zassert_equal(0, at_params_string_put(&test_list, 1,
+					      test_str, sizeof(test_str)),
+		      "String put should return 0");
+
+	zassert_equal(-EINVAL, at_params_string_get(NULL, 1,
+						    test_buf, &test_buf_len),
+		      "String get should return -EINVAL");
+
+	zassert_equal(-EINVAL, at_params_string_get(&test_list, 1,
+						    NULL, &test_buf_len),
+		      "String get should return -EINVAL");
+
+	zassert_equal(-EINVAL, at_params_string_get(&test_list, 0,
+						    test_buf, &test_buf_len),
+		      "String get should return -EINVAL");
+
+	test_buf_len = 0;
+	zassert_equal(-ENOMEM, at_params_string_get(&test_list, 1,
+						    test_buf, &test_buf_len),
+		      "String get should return -ENOMEM");
+
+	test_buf_len = sizeof(test_buf);
+	zassert_equal(0, at_params_string_get(&test_list, 1,
+					      test_buf, &test_buf_len),
+		      "String get should return 0");
+
+	zassert_equal(sizeof(test_str), test_buf_len,
+		      "test_buf_len should be equal to sizeof(test_str)");
+
+	zassert_equal(0, memcmp(test_str, test_buf, sizeof(test_str)),
+		      "test_str and test_buf should be equal");
+}
+
+static void test_params_put_get_string_teardown(void)
+{
+	at_params_list_free(&test_list);
+}
+
+static void test_params_put_get_array_setup(void)
+{
+	at_params_list_init(&test_list, TEST_PARAMS);
+	at_params_int_put(&test_list, 0, 1);
+}
+
+static void test_params_put_get_array(void)
+{
+	const u32_t test_array[] = {1, 2, 3, 4, 5, 6, 7};
+	u32_t       test_buf[32];
+	size_t      test_buf_len = sizeof(test_buf);
+
+	zassert_equal(-EINVAL, at_params_array_put(NULL, 1,
+					test_array, sizeof(test_array)),
+		      "Array put should return -EINVAL");
+
+	zassert_equal(-EINVAL, at_params_array_put(&test_list, 1,
+					NULL, sizeof(test_array)),
+		      "Array put should return -EINVAL");
+
+	zassert_equal(0, at_params_array_put(&test_list, 1,
+					      test_array, sizeof(test_array)),
+		      "Array put should return 0");
+
+	zassert_equal(-EINVAL, at_params_array_get(NULL, 1,
+						    test_buf, &test_buf_len),
+		      "Array get should return -EINVAL");
+
+	zassert_equal(-EINVAL, at_params_array_get(&test_list, 1,
+						    NULL, &test_buf_len),
+		      "Array get should return -EINVAL");
+
+	zassert_equal(-EINVAL, at_params_array_get(&test_list, 0,
```

Comment:
You could add a test to check when the output parameters (array, len) are NULL.

Reply:
Same as above.